### PR TITLE
Split up tutorials into several pages, Closes #252

### LINKED ISSUE #252: Split up example app tutorials
To improve overview, the tutorials should be on separate pages.
- improve verification in last step: first check logsd to see tx is in tx pool. then note down the next block id in logs (probably included the tx), then make db query to verify the tx was included in the block

### PR DESCRIPTION
**Pull request by Tschakki**
Closes #252

### CODE REVIEW

File: build-blockchain-app.md [hunk 1]
```diff
@@ -46,10 +46,10 @@ For comparison, please view the [list of all transaction types](../lisk-protocol
 
 To create custom transaction types, utilize the provided `BaseTransaction` interface as described in the section for [custom transactions](custom-transactions.md).
 
-See the [Examples page](examples.md) for creating custom transaction types and how to add them to your application:
+See the [tutorials](tutorials.md) to see examples of how to create custom transaction types and how to add them to your application:
 
-- [Hello World App](examples.md#hello-world-app): A very basic example of a Lisk blockchain application with 1 custom transaction type.
-- [Cashback App](examples.md#hello-world-app): A simple example blockchain app, that implements the new transaction type `Cashback`.
+- [Hello World App](tutorials/hello-world.md): A very basic example of a Lisk blockchain application with 1 custom transaction type.
```
Comment (RachBLondon) [inline build-blockchain-app.md]:
```suggestion
- [Hello World App](tutorials/hello-world.md): A example of a Lisk blockchain application with 1 custom transaction type, which extends the BaseTransaction type.
```
I have changed the wording, to make not appear that it is more simple than the Cashback transaction.

File: build-blockchain-app.md [hunk 2]
```diff
@@ -46,10 +46,10 @@ For comparison, please view the [list of all transaction types](../lisk-protocol
...
+- [Cashback App](tutorials/cashback.md): A simple example blockchain app, that implements the new transaction type `Cashback`.
```
Comment (RachBLondon) [inline build-blockchain-app.md]:
```suggestion
- [Cashback App](tutorials/cashback.md): A more practical example creating the new transaction type `Cashback`. Cashback extends the pre-existing transaction type TransferTransaction, making for a straightforward implementation.
```
Re-worded to give some more context to the tutorial

File: tutorials/hello-world.md [hunk 3]
```diff
@@ -287,7 +313,7 @@ npx pm2 stop hello # stop the hello app
 npx pm2 start hello # start the hello app
 ```
 
-### 6. Interact with the network
+## 6. Interact with the network
```
Comment (RachBLondon) [inline tutorials/hello-world.md:338]: Not part of the changes in the PR - but do we need to import 
```
const { validateAddress, validatePublicKey } = require('@liskhq/lisk-validator');
const { getAddressFromPublicKey } = require('@liskhq/lisk-cryptography');
```
on line 332 and 333? I can't see where they are used in the code snippett.

File: tutorials/cashback.md [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+npx pm2 start cashback # start the cashback app
+```
+
+## 6. Interact with the network
+
+Now that the network is started, let's try to send a `CashbackTransaction` to our node to see if it gets accepted.
+
+As first step, create the transaction object.
+
+First, let's reuse the script [create_sendable_transaction.js](https://github.com/LiskHQ/lisk-sdk-examples/blob/development/scripts/create_sendable_transaction_base_trs.js) which we already described in [step 6 of Hello World app](#6-interact-with-the-network).
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
First, let's reuse the script [create_sendable_transaction.js](https://github.com/LiskHQ/lisk-sdk-examples/blob/development/scripts/create_sendable_transaction_base_trs.js) which was created in step 6 of the other example [Hello World App](#6-interact-with-the-network).
```

**Review by RachBLondon — DISMISSED** [review]

File: tutorials.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+Here we provide you with a detailed step-by-step tutorials that guide you through the process of developing blockchain application with the Lisk SDK.
+
+All tutorials cover the following steps:
+
+- how to initially set up the Lisk SDK and get the network started.
+- how to develop a new transaction type and how to register it with the application.
+- how to create a sendable transaction object of the new type, and send it to the network.
+
+## Hello World
+A hello world application of the alpha SDK.
```
Comment (RachBLondon) [inline tutorials.md]:
```suggestion
A hello world application built using the alpha SDK.
```

File: tutorials.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+Explains how to create a new transaction type from scratch.
+
+A simple example to showcase how to create a new transaction type by implementing the [`BaseTransaction` interface](custom-transactions.md).
+
+- [Full code example of Hello World app](https://github.com/LiskHQ/lisk-sdk-examples/tree/development/hello_world) on Github.
+- [Hello World Tutorial](tutorials/hello-world.md)
+
+## Cashback
+
+The example of a simple but valid use-case - cashback.
```
Comment (RachBLondon) [inline tutorials.md]:
```suggestion
The example of a simple but valid use-case - a cashback transaction.
```

File: tutorials.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+On top of the regular balance transfer, the example implements the cashback functionality.
```
Comment (RachBLondon) [inline tutorials.md]:
```suggestion
Built by extending the balance transfer transaction, in addition to the regular transfer functionality, the cashback transaction also applies an inflationary 10% to senders account.
```

File: tutorials.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+- [Full code example of Hello World app](https://github.com/LiskHQ/lisk-sdk-examples/tree/development/hello_world) on Github.
+- [Hello World Tutorial](tutorials/hello-world.md)
+
+## Cashback
+
+The example of a simple but valid use-case - cashback.
+On top of the regular balance transfer, the example implements the cashback functionality. 
+
+Explains how to create a new transaction type that inherits from an already existing transaction type.
```
Comment (RachBLondon) [inline tutorials.md]:
```suggestion
This example shows you how to create a new custom transaction inheriting from a pre-existing transaction type.
```

File: tutorials.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+- [Full code example of Cashback app](https://github.com/LiskHQ/lisk-sdk-examples/tree/development/cashback) on Github.
```
Comment (RachBLondon) [inline tutorials.md]:
```suggestion
- [A full example Cashback application's codebase ](https://github.com/LiskHQ/lisk-sdk-examples/tree/development/cashback) is available Github.
```

File: tutorials.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+- [Cashback Tutorial](tutorials/hello-world.md)
```
Comment (RachBLondon) [inline tutorials.md]:
```suggestion
- [Cashback Tutorial](tutorials/cashback.md)
```

File: tutorials/cashback.md [hunk 11]
```diff
@@ -0,0 +1,462 @@
+# Cashback App
+
+A simple application which rewards its users for sending tokens. 
+
+The Cashback App implementation goes as following:
+
+- __Steps 1-5__ describe what needs to be implemented on the server-side of the blockchain application.
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
- __Steps 1-5__ describes the sever-side implementation of the blockchain application.
```

File: tutorials/cashback.md [hunk 12]
```diff
@@ -0,0 +1,462 @@
...
+- __Step 6__ explains how to interact with the network from the client-side.
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
- __Step 6__  shows how to interact with the network from the (blockchain) client-side using a node script.
```

File: tutorials/cashback.md [hunk 13]
```diff
@@ -0,0 +1,462 @@
...
+
+> Check out the __full code example__ for the [Cashback App on Github](https://github.com/LiskHQ/lisk-sdk-test-app/cashback).
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
> Check out the __full code example__ of the [Cashback App on Github](https://github.com/LiskHQ/lisk-sdk-test-app/cashback).
```

File: tutorials/cashback.md [hunk 14]
```diff
@@ -0,0 +1,462 @@
...
+
+As next step, we want to install the `lisk-sdk` package and add it to our projects' dependencies.
+Before installing it, make sure to follow the instructions in the [Lisk SDK - Pre-Install](../lisk-sdk/introduction.md#pre-installation) section.
+
+```bash
+npm install --save lisk-sdk@alpha # install lisk-sdk as dependency for the server side
+npm install --save @liskhq/validator @liskhq/cryptography # install lisk-elements dependencies for the client side scripts
+```
+
+Create 2 folders `client` and `server`, which will hold the corresponding scripts for the blockchain application.
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
Create two folders `client` and `server`, which will hold the corresponding scripts for the blockchain application.
```

File: tutorials/cashback.md [hunk 15]
```diff
@@ -0,0 +1,462 @@
...
+
+- the `server` folder holds all the code that is needed to start a node and connect it to a network.
+- the `client` folder holds scritps that communicate with the network through the API of the node.
+
+```bash
+mkdir client server # create the 2 folder inside the root directory of your blockchain application
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
mkdir client server # creates the folder inside the root directory of your blockchain application
```

File: tutorials/cashback.md [hunk 16]
```diff
@@ -0,0 +1,462 @@
...
+
+Inside the `server` folder, create the index file of your blockchain application:
+
+```bash
+touch index.js
+```
+
+## 2. Configure the application
+
+Next, let's configure the application, to provide basic information about the app we are going to build:
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
Next, let's configure the application, to provide basic information about the application we are going to build:
```

File: tutorials/cashback.md [hunk 17]
```diff
@@ -0,0 +1,462 @@
...
+
+```js
+//server/index.js
+const { Application, genesisBlockDevnet, configDevnet } = require('lisk-sdk'); // require application class, the default genesis block and the default config for the application
+
+const app = new Application(genesisBlockDevnet, configDevnet); // create the application instance
+```
+> *See the complete file on Github: [cashback/server/index.js](https://github.com/LiskHQ/lisk-sdk-test-app/tree/development/cashback/server/index.js).*
+
+In the `line 2`, we require the needed dependencies from the `lisk-sdk` package.
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
On `line 2`, we require the needed dependencies from the `lisk-sdk` package.
```

File: tutorials/cashback.md [hunk 18]
```diff
@@ -0,0 +1,462 @@
...
+The most important one is the `Application` class, which is used in `line 4` to create the application instance.
+The application instance will start the whole application at the bottom of `index.js`.
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
The application instance is used to start the blockchain application at the bottom of `index.js`.
```

File: tutorials/cashback.md [hunk 19]
```diff
@@ -0,0 +1,462 @@
...
+
+In `line 4` , the application instance gets initialized.
+By passing the parameters for the [genesis block](../lisk-sdk/configuration.md) and the [configuration template](https://github.com/LiskHQ/lisk-sdk/blob/development/sdk/src/samples/config_devnet.json), the application is configured with most basic configurations to start the node.
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
By passing the parameters for the [genesis block](../lisk-sdk/configuration.md) and the [configuration template](https://github.com/LiskHQ/lisk-sdk/blob/development/sdk/src/samples/config_devnet.json), the application is configured with the most basic configurations to start the node.
```

File: tutorials/cashback.md [hunk 20]
```diff
@@ -0,0 +1,462 @@
...
+
+## 3. Create a new transaction type
+
+Now, we want to create a new [custom transaction type](custom-transactions.md) `CashbackTransaction`: 
+It extends the pre-existing transaction type `TransferTransaction`.
+In difference to the normal `TransferTransaction`, the Cashback transaction type pays out a 10% bonus reward to the sender of any Cashback transaction type.
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
The difference between the regular `TransferTransaction` and the `CashbackTransaction`, is that Cashback transaction type also pays out a 10% bonus reward to its sender.
```

File: tutorials/cashback.md [hunk 21]
```diff
@@ -0,0 +1,462 @@
...
+
+So e.g. if Alice sends 100 token to Bob as a Cashback transaction, Bob would receive the 100 token and Alice would receive additional 10 tokens as a cashback.
+
+> If you compare the methods below with the methods we implemented for the `HelloTransaction`, you will notice, that we implement less methods for the `CashbackTransaction`.
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
> If you compare the methods below with the methods we implemented in the `HelloTransaction`, you will notice, that we implement fewer methods for the `CashbackTransaction`.
```

File: tutorials/cashback.md [hunk 22]
```diff
@@ -0,0 +1,462 @@
...
+> This is because we extend the `CashbackTransaction` from an already existing transaction type `TransferTransaction`.
+> As a result, all required methods are implemented already inside the `Transfertransaction` class, and we only need to overwrite/extend explicitely the methods we want to customize.
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
> As a result, all required methods are implemented already inside the `TransferTransaction` class, and we only need to overwrite/extend explicitely the methods we want to customize.
```

File: tutorials/cashback.md [hunk 23]
```diff
@@ -0,0 +1,462 @@
...
+
+```bash
+node index.js | npx bunyan -o short
+```
+
+`node index.js` will start the node, and `| npx bunyan -o short` will pretty-print the logs in the console.
+
+Check the logs, to verify the network has started successfully.
+
+If something went wrong, the process should stop and an error with debug information is displayed.
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
If an error occurs, the process should stop and the error with debug information will be displayed.
```

File: tutorials/cashback.md [hunk 24]
```diff
@@ -0,0 +1,462 @@
...
+
+If everything is ok, you should be able to see the following console logs:
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
If everything is ok, the following logs will be displayed:
```

File: tutorials/cashback.md [hunk 25]
```diff
@@ -0,0 +1,462 @@
...
+14:01:40.621Z  INFO lisk-framework: Forging enabled on account: 12254605294831056546L
+14:01:40.624Z  INFO lisk-framework: Forging enabled on account: 14018336151296112016L
+14:01:40.627Z  INFO lisk-framework: Forging enabled on account: 2003981962043442425L
+[...]
+```
+
+
+## 6. Interact with the network
+
+Now that the network is started, let's try to send a `CashbackTransaction` to our node to see if it gets accepted.
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
Now that your network is running, let's try to send a `CashbackTransaction` to our node to see if it gets accepted.
```

File: tutorials/cashback.md [hunk 26]
```diff
@@ -0,0 +1,462 @@
...
+
+As first step, create the transaction object.
+
+First, let's reuse the script [create_sendable_transaction_base_trs.js](https://github.com/LiskHQ/lisk-sdk-examples/blob/development/scripts/create_sendable_transaction_base_trs.js) which we already described in [step 6 of Hello World app](#6-interact-with-the-network).
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
First, let's reuse the script [create_sendable_transaction_base_trs.js](https://github.com/LiskHQ/lisk-sdk-examples/blob/development/scripts/create_sendable_transaction_base_trs.js) which we already described in [step 6 of Hello World app](/hello-world.md#6-interact-with-the-network).
```
👆 Not sure if that link is correct, but I think it needs changing to point to the Hello World tutorial.

File: tutorials/cashback.md [hunk 27]
```diff
@@ -0,0 +1,462 @@
...
+
+We can use this function in our script for printing a sendable `CashbackTransaction` object:
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
We can call the `createSendableTransaction` function to print a sendable `CashbackTransaction` object:
```

File: tutorials/cashback.md [hunk 28]
```diff
@@ -0,0 +1,462 @@
...
+   "fee":"10000000",
+   "signature":"0a3f41cc529f9de523cadc7db64e9436014d1b10ca2158bbce0469e8e76dfd021358496440da43acaf64d0223d3514609fc1aa41646be56353207d88a03b1305",
+   "signatures":[],
+   "asset":{}
+}
+```
+
+Now that we have a sendable transaction object, let's send it to our node and see how it gets processed by analyzing the logs.
+
+For this, we utilize the API of the node and post the created transaction object to the transaction endpoint of the API.
```
Comment (RachBLondon) [inline tutorials/cashback.md]:
```suggestion
For this, we utilize the http API of the node and post the created transaction object to the transaction endpoint of the API.
```

**Review by RachBLondon — CHANGES_REQUESTED** [review]

File: tutorials/cashback.md (hunk 26 above)
Comment (Tschakki) [reply tutorials/cashback.md]: @RachBLondon: This part of docs might be outdated soon, if this branch is merged: https://github.com/LiskHQ/lisk-sdk/compare/release/2.1.0...3739-create_sendable

I hope it gets merged today :)

**Review by RachBLondon — APPROVED** [review]

**Pull request merged**